File: calculator.cpp
#include <iostream>
using namespace std;

int get_sum() {
    int sum = 0;
    int count;
    cin >> count;
    while(count--) {
        int num;
        cin >> num;
        sum += num;
    }
    return sum;
}

int get_product() {
    int product = 1;
    int count;
    cin >> count;
    while(count--) {
        int num;
        cin >> num;
        product *= num;
    }
    return product;
}

int main() {
    string cmd;
    while(true) {
        cin >> cmd;

        if (cmd == "exit") {
            break;
        } else if (cmd == "sum") {
            cout << get_sum() << endl;
        } else if (cmd == "product") {
            cout << get_product() << endl;
        } else {
            cerr << "unknow command: " + cmd << endl;
        }

    }
    return 0;
}

File: calculator.py
#!/usr/bin/python
import sys
import os
sys.stdout = os.fdopen(sys.stdout.fileno(), 'w', 0)

while(True) :
    cmd = raw_input()
    if (cmd == "exit"):
        break
    elif (cmd == "sum"):
        count = int(raw_input())
        _sum = 0
        def add_to_sum(num):
            global _sum
            _sum += num
        map(add_to_sum,
            [int(raw_input()) for i in range(0, count)])
        print _sum
    else:
        sys.stderr.write("unknown command! %s \n" % (cmd,))
        break;
